v2_1/ws/rest/docs/4_8_maintenance.md

```diff
@@ -0,0 +1,490 @@
+
+The `PUT` method may be used like this:
+
+- under `/structure/{maintainable}/{identifier}` for the Maintainable Artefact of type `{maintainable}`, identified by `{identifier}`, i.e. an SDMX DSD message with `ESTAT:NA_MAIN(1.1)` under `/structure/datastructure/ESTAT/NA_MAIN/1.1`
+
+The result, following the current practices of updates in SDMX, is to **completely replace** the identified Artefact.
+
+##### Client
+
+In order to update Artefact(s), the client:
+
+- MAY set the `Accept` header to indicate the preferred response format;
+- MUST set the `Content-type` header according to the format of the submitted Artefact(s);
+- MUST include in the request body, one or more Maintainable Artefacts in the SDMX format indicated in the `Content-type` header and of the SDMX type indicated in the resource, i.e.:
+- for `POST`:
+  - any set of Maintainable Artefacts under resource `/structure/`
+  - one or more Maintainable Artefacts of the specific type under the corresponding resource type, e.g. Dataflows under `/structure/dataflow/`
+- for `PUT`:
+  - one Maintainable Artefact, of the specific type under the corresponding resource type and identified according to the resource parameters, e.g., DSD `ESTAT:NA_MAIN(1.1)` under `/structure/datastructure/ESTAT/NA_MAIN/1.1`
+
+##### Server
+
+In response to Artefact(s) update, the server:
+
+- MUST respond with `200` in case of successful update;
+- MUST return a `SubmitStructureResponse` message with the result of the action(s), according to the `Accept` header, or the default, if the `Accept` type is not supported (currently available only in SDMX-ML);
+- MUST set the `Content-type` according to the returned format;
+- MUST respond with `422` in case of resource type mismatch, i.e. the resource type identified in the URL does not match the Artefact type(s) of the included SDMX Artefact(s), or the identification of the Artefact (in case of `PUT`) does not match the resource path parameters.
+- MUST respond with `404` if the resource was not found;
+
+##### Response
+
+The `SubmitStructureResponse` message must be returned in any case (success, partial success, failure).
+This is defined as part of the `RegistryInterface` messages.
+The details of the message are explained in section **Structure response message**, below.
+
+The following matrix summarises the returned `HTTP` response codes.
+
+| Method | Exists | Versioning respected | Is Referenced | References exist/provided | Return Code |
+|---|---|---|---|---|---|
+| PUT/POST | T | T | F | T | `200` (successful) or `207` (partially successful)  |
+| PUT/POST | T | T | F | F | `409` |
+| PUT/POST | T | T | T | T | `200` if update is possible - `409` if references would break |
+| PUT/POST | T | F | - | - | `409` in case of changes that violate the versioning requirements |
```

Comment:
So, in case semantic versioning is used and the artefact cannot be updated, an attempt to update that artefact will trigger a 409?

Reply:
Yes, it would mean a conflict in the submission.